packages/stencil-library/src/components/alert/alert.e2e.ts

```diff
@@ -1,10 +1,119 @@
-import { newE2EPage } from '@stencil/core/testing';
+import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';
+
+const renderAndGetAlertDiv = async (content: string): Promise<{ alertDiv: E2EElement; page: E2EPage }> => {
+  const page = await newE2EPage();
+  await page.setContent(content);
+  const alertDiv = await page.find('.usa-alert');
+  return { alertDiv, page };
+};
 
 describe('<njwds-alert>', () => {
-  it('should render an njwds-alert component', async () => {
-    const page = await newE2EPage();
-    await page.setContent(`<njwds-alert><span>Body alert text</span></njwds-alert`);
-    const alertComponent = await page.find('njwds-alert');
-    expect(alertComponent).not.toBeNull();
+  it('should render an njwds-alert component where default slotted content displays as alert text', async () => {
+    const { alertDiv } = await renderAndGetAlertDiv(`
+      <njwds-alert>
+        <span>Body alert text</span>
+      </njwds-alert>
+    `);
+
+    expect(alertDiv).not.toBeNull();
+    const slotContainer = await alertDiv.find('p');
+    expect(slotContainer).toHaveClass('usa-alert__text');
+    const slottedContent = await slotContainer.find('span');
+    expect(slottedContent).toEqualText('Body alert text');
+  });
+
+  it('applies no-icon class when no-icon is true', async () => {
+    const { alertDiv } = await renderAndGetAlertDiv(`
+      <njwds-alert no-icon>
+        <span>Body alert text</span>
+      </njwds-alert>
+    `);
+    expect(alertDiv).toHaveClass('usa-alert--no-icon');
```

Comment:
[pebble] I think for this test to be robust, I'd like a matching test that confirms it does NOT have the no-icon class when no-icon is false. 

I think these tests would all still be passing if I were to just add the no-icon class to the div all of the time regardless of the prop value.